platform/darwin/src/MGLFeature.mm

```diff
@@ -37,11 +49,30 @@ @implementation MGLPolylineFeature
 
 @synthesize identifier;
 @synthesize attributes;
+// 
 
 - (id)attributeForKey:(NSString *)key {
     return self.attributes[key];
 }
 
+- (NS_DICTIONARY_OF(NSString *, id) *)featureDictionary {
+    
+    NSMutableArray *coordinates = [NSMutableArray array];
+    
+    for (int index = 0; index < self.pointCount; index++) {
```

Comment:
int might not be large enough for the number of coordinates that may be in this polyline. Use the same type as pointCount.